libexec/scoop-shim.ps1

```diff
@@ -0,0 +1,211 @@
+if ($err) { abort "scoop shim: $err" 1 }
+$global = $opt.g -or $opt.global
+$globalTag = if ($global) { 'global' } else { 'local' }
+
+if ($SubCommand -notin @('add', 'remove', 'rm', 'list', 'info', 'alter')) {
+    'ERROR: <subcommand> must be one of: add, remove/rm, list, info, alter'
+    my_usage
+    exit 1
+}
+
+if (!$ShimName -and $SubCommand -ne 'list') {
+    'ERROR: <shim_name> must be specified'
+    my_usage
+    exit 1
+}
+
+if ($SubCommand -eq 'add' -and $addArgs.Length -eq 0) {
+    'ERROR: <command_path> must be specified'
+    my_usage
+    exit 1
+}
+
+if (-not (Get-FormatData ScoopShim)) {
+    Update-FormatData "$PSScriptRoot\..\supporting\formats\ScoopTypes.Format.ps1xml"
+}
+
+$localShimDir = shimdir $false
+$globalShimDir = shimdir $true
+
+function Get-ShimInfo($ShimPath) {
+    $info = @{ PSTypeName = 'ScoopShim' }
+    $info.Name = strip_ext (fname $ShimPath)
+    $info.Path = $ShimPath -replace 'shim$', 'exe'
+    $info.Source = get_app_name_from_shim $ShimPath
+    $info.Type = if ($ShimPath.EndsWith('.shim')) { 'Application' } elseif ($ShimPath.EndsWith('.cmd')) { 'Script' } else { 'Unknown' }
+    $altShims = Get-Item -Path "$ShimPath.*" -Exclude '*.shim', '*.cmd', '*.ps1'
+    if ($altShims) {
+        $info.Alternatives = @($info.Source) + ($altShims | ForEach-Object { $_.Extension.Remove(0, 1) } | Select-Object -Unique)
+    }
+    $info.IsGlobal = if ($ShimPath.StartsWith("$globalShimDir")) { $true } else { $false }
+    $info.IsHidden = if ((Get-Command -Name $info.Name).Path -eq $info.Path) { $false } else { $true }
+    [PSCustomObject]$info
+}
+
+function Get-ShimPath($ShimName, $Global) {
+    '.shim', '.cmd' | ForEach-Object {
+        $shimPath = Join-Path (shimdir $Global) "$ShimName$_"
+        if (Test-Path -Path $shimPath) {
+            return $shimPath
+        }
+    }
+}
+
+function Get-ShimTarget($ShimPath) {
+    $shimTarget = if ($ShimPath.EndsWith('.shim')) {
+        (Get-Content -Path $ShimPath | Select-Object -First 1).Replace('path = ', '')
+    } else {
+        ((Select-String -Path $ShimPath -Pattern '^(?:@rem|#)\s*(.*)$').Matches.Groups | Select-Object -Index 1).Value
+    }
+    if (!$shimTarget) {
+        $shimTarget = ((Select-String -Path $ShimPath -Pattern '[''"]([^@&]*?)[''"]' -AllMatches).Matches.Groups | Select-Object -Last 1).Value
+    }
+    $shimTarget | Convert-Path
+}
+
+switch ($SubCommand) {
+    'add' {
+        if ($addArgs[0] -match '[\\/]') {
+            $commandPath = $addArgs[0]
+        } else {
+            $commandPath = Get-ShimTarget (Get-ShimPath $addArgs[0] $global)
+        }
+        if ($commandPath -and (Test-Path $commandPath)) {
+            Write-Host "Adding $globalTag shim " -NoNewline
+            Write-Host $ShimName -ForegroundColor Cyan -NoNewline
+            Write-Host ' ... ' -NoNewline
+            shim $commandPath $global $ShimName $(if ($addArgs.Length -gt 1) { $addArgs[1..($addArgs.Length - 1)] })
```

Comment:
It will be a very common use case where users can specify shim args. But there is an error. See this example:

![image](https://user-images.githubusercontent.com/46838874/154116469-869939b6-3e67-4b2a-b2a4-510dcca2cac4.png)

Here the scoop-shim.ps1 script is swallowing up the args needed to pass into the `add` subcommand.

Reply:
Hmm, you should put arguments insides <command_path>, I'll change the code

Reply:
Could you give an example (specifically when the app binary path has spaces in it)?

Reply:
See https://github.com/ScoopInstaller/Scoop/pull/4736#issuecomment-1041571435